### ISSUE TITLE
Lazy artifact without unpacking (non-tarball)

### ISSUE THREAD

**Issue by jonas-schulze** [JuliaLang/Pkg.jl#2764]
I would like to "deliver" some `mat` data set that I need during package testing as an artifact. The data set happened to be hosted already, though as is and not as a `tar.gz`. IIRC `.mat` support compression on their own, so wrapping them in a `tar.gz` feels odd. 

How do I declare a `lazy` artifact (containing only a single file) that doesn't need to be unpacked?

If that's not possible (yet), I would like to propose to add a new keyword `unpack` (default: `true` which matches the current behavior) to the `Artifacts.toml`.

Somewhat related:
* https://github.com/JuliaLang/Pkg.jl/issues/1467 (suggested to me that unpacking does not always happen)
* https://github.com/JuliaLang/Pkg.jl/issues/1950 (would need the keyword as well)
* https://discourse.julialang.org/t/creating-artifacts-toml-for-existing-tarball/33365

**Comment by DilumAluthge** [comment]
The DataDeps.jl package (https://github.com/oxinabox/DataDeps.jl) might be a good solution for your use case.

**Comment by jonas-schulze** [comment]
Ref https://github.com/oxinabox/DataDeps.jl/issues/113

**Comment by KristofferC** [comment]
> Another layer of compression shouldn't really hurt though

But then you need to rehost the files.

**Comment by simonbyrne** [comment]
Agreed, there are many dataset hosting providers which expect you to upload the file directly, rather than uploading a tarball wrapping a file.

**Comment by jonas-schulze** [comment]
> But then the artifact isn't actually content-addressed anymore: you need to know the content hash _and_ the path inside of the directory, [...]

Isn't this exactly what is required now already from a user's perspective? In order to access anything from an artifact, the user has to `joinpath(artifact"foo", "data.mat")`. Here, `artifact"foo"` resolves to the content-addressed hash of the directory and `data.mat` is the object within the user is actually interested in.

> [...] and if two different artifact files referenced the same content address with a different hash, then they could extract the data to a different location.

I think I don't quite understand what you mean. If two artifact files (does this refer to "descriptors", i.e. `artifact"foo"` and `artifact"bar"`?) refer to the same content, they will by design resolve to the same hash, won't they?

The considerations you described sound more like implementation details to me -- no offend. All I am asking for is an option to skip a certain part of the download/registration/creation process of an artifact, namely archive inflation. I am not questioning what an artifact is. An artifact remains a single file before and during download (a compressed or un-compressed tar-ball, or an arbitrary file) which becomes a content-addressed directory. This doesn't change at all. And from a user's perspective it doesn't change either. The user shouldn't need to care how the content-hash comes to be, because a user never gets in touch with it anyway. This is a detail hidden within `artifact"foo"`, as it should be.
